addons/html_builder/static/src/website_builder/plugins/form/form_option.xml

```diff
@@ -85,6 +85,9 @@
 </t>
 
 <t t-name="html_builder.website.s_website_form_field_option">
+    <t t-if="domState.el.classList.contains('s_website_form_model_required')">
```

Comment:
You should define a key in the useDomState. 
With the current solution, you won't re-render your template if the class is added or removed.